keanu-project/src/main/java/io/improbable/keanu/network/NetworkState.java

```diff
@@ -19,4 +19,6 @@
     <T> T get(VertexId vertexId);
 
     Set<VertexId> getVertexIds();
+
+    double getLogOfMasterP();
```

Comment:
I see your point. It doesn't quite fit in that `logOfMasterP` is included in the "State" as it could be derived. My reasoning for this approach was that when you sample in a batch based fashion you do get the `logOfMasterP` for each sample. 

An alternative would be to calculate the `logOfMasterP` after streaming samples. By either:
 - Adding a method to `BayesianNetwork` that takes a `List<NetworkState>` and returns a list of `logOfMasterP`. Then changing the creation to `NetworkSamples.from(List<NetworkState>, List<logOfMasterP(Double)>)`.
 - Changing the creation to `NetworkSamples.from(List<NetworkState>, BayesianNetwork)`. Then in this method getting the `logOfMasterP` for each state.

What are your thoughts?

Reply:
Another option would for the stream to be of a new class that includes the `NetworkState` and `logOfMasterP`. This would stop muddying the `NetworkState` but does add another class into the mix.

Reply:
Spoke to Caleb about this. Going to create a class called `NetworkSample` which is the composition of a `NetworkState` and `logOfMasterP`. This can be further extended to include any other information relevant for that sample.

Reply:
As discussed offline - won't adding an extra class in the chain mean a lot more allocations (and thus impact performance a reasonable amount?).  This might not be something we want to worry about at the moment, but adding an extra class in the hot path could be a bad thing?